openapi.yaml

```diff
@@ -64,21 +64,104 @@ components:
         default: 25
       example: 15
   schemas:
-    Make:
-      description: Represents the data for the Make entity.
+    ListingFacetsDto:
+      description: Represents the request structure for listing facets.
+      required:
+        - facets
+      type: object
+      properties:
+        query:
+          $ref: '#/components/schemas/ListingQueryDto'
+        facets:
+          description: Facets to retrieve.
+          type: array
+          items:
+            $ref: '#/components/schemas/ListingFacet'
+    ListingQueryDto:
+      description: Represents the query parameters for filtering listings.
+      type: object
       properties:
-        key:
-          description: Unique key of the make.
+        fuelTypes:
+          description: Filter listings by fuel types.
+          uniqueItems: true
+          type: array
+          items:
+            type: string
+            enum:
+              - cng-petrol
+              - diesel
+              - electric
+              - ethanol-petrol
+              - hev-diesel
+              - hev-petrol
+              - hydrogen
+              - lpg-petrol
+              - mhev-diesel
+              - mhev-petrol
+              - petrol
+              - phev-diesel
+              - phev-petrol
+              - two-stroke-mixture
+        makeModelVersions:
+          description: Filter listings by make and model combinations.
+          uniqueItems: true
+          type: array
+          items:
+            $ref: '#/components/schemas/ListingQueryDto$MakeModelVersion'
+        vehicleCategories:
+          description: Filter listings by vehicle categories.
+          uniqueItems: true
+          type: array
+          items:
+            type: string
```

Comment:
should we extract to $ref: '#/components/schemas/VehicleCategory' for other endpoints to reuse?

Reply:
Let's extract it once we have it in multiple locations